NanoGardener/python/modules/BWEwkSingletReweighter.py

```diff
@@ -595,29 +598,28 @@ def analyze(self, event):
             for line in self.shifts:
               if float(line[0])==self.mH and float(line[1])==relw and float(line[2])==-1:
                 shift = float(line[3])
-                break
+                break      
           weights[name] = (1./shift)*decayWeight*CPSweight
-          self.mela.setMelaHiggsMassWidth(self.mH, self.gsm)
+          self.mela.setMelaHiggsMassWidth(self.mH, gprime)
           self.mela.setupDaughters((self.productionProcess=="VBF"), int(ids[0]), int(ids[1]), int(ids[2]), int(ids[3]),
                                                        fourMomenta[0], fourMomenta[1], fourMomenta[2], fourMomenta[3],
                                                        partons, partonIDs,
                                                        mothers, motherIDs)
-          addweight = {}
-          addweight["_I"] = self.mela.weightStoI()
-          #addweight["_I_Honly"] = self.mela.weightStoI_H()
-          #addweight["_I_Bonly"] = self.mela.weightStoI_B()
-          #addweight["_I_HB"] = self.mela.weightStoI_HB()
-          addweight["_B"] = self.mela.weightStoB()
-          #addweight["_H"] = self.mela.weightStoH()
-
-          for appendix in self.branches:
-            if math.isnan(addweight[appendix]) or math.isinf(addweight[appendix]): #dirty protection for occasional failures
-              self.NANinfo
-              addweight[appendix]=0.
-            weights[name+appendix] = weights[name]*addweight[appendix]
-
-          for appendix in [""]+self.branches:
-            self.out.fillBranch(name+appendix, weights[name+appendix])
+          #addweight = {}
+          weights[name+"_I"] = math.sqrt(weights[name])*self.mela.weightStoI()
+          weights[name+"_I_Honly"] = math.sqrt(weights[name])*self.mela.weightStoI_H()
+          weights[name+"_I_Bonly"] = math.sqrt(weights[name])*self.mela.weightStoI_B()
+          # now reset to the SM high mass signal and do not touch the normalization again
+          self.mela.setMelaHiggsMassWidth(self.mH, self.gsm)
+          weights[name+"_I_HB"] = self.mela.weightStoI_HB()
+          # reset to SM width before reweighting to H and B
+          self.mela.setMelaHiggsMassWidth(self.mH, self.gsm)
```

Comment:
Was this line supposed to be different from the one in line 613? (Same for lines 563/566)

Reply:
No, I think is should be the same, I'll remove one occurrence. Thanks!